2.0/s2i/bin/run

```diff
@@ -1,5 +1,26 @@
 #!/bin/bash
 
+function export_memory_limit() {
+  if [ -f /sys/fs/cgroup/memory/memory.limit_in_bytes ]; then
+    export MEMORY_LIMIT_IN_BYTES=$(cat /sys/fs/cgroup/memory/memory.limit_in_bytes)
+    return 0
+  fi
+  return 1
+}
+
+function export_number_of_cores() {
+  if [ -f /sys/fs/cgroup/cpuset/cpuset.cpus ]; then
+    export NUMBER_OF_CORES=`awk ' \
+      BEGIN { i = 0; FS="-"; RS="[,\n]" } \
+      /^[0-9]+-[0-9]+$/ { i += $2-$1+1 }  \
+      /^[0-9]+$/        { ++i }           \
+      END { print i} \
+      ' /sys/fs/cgroup/cpuset/cpuset.cpus`
+    return 0
+  fi
+  return 1
+}
```

Comment:
@mnagy you have to either check for the return 1 or the `NUMBER_OF_CORES` ;-) also the user of this function will need to know what variable it is exporting, since you hide it in the body (which is not the best api design ;-)